libclamav/readdb.c

```diff
@@ -2943,6 +2943,30 @@ static int cli_loadcdb(FILE *fs, struct cl_engine *engine, unsigned int *signo,
     return CL_SUCCESS;
 }
 
+static cl_error_t setHash(const char *const token, uint8_t hashDest[SHA1_HASH_SIZE],
+                          const char *const varname, uint32_t line)
```

Comment:
Are these supposed to be `const char *const` or just `const char*`?   I'm thinking the latter.

Also, github's static analysis tool is claiming an issue with the `cli_hex2str_to` call: https://lgtm.com/projects/g/Cisco-Talos/clamav/rev/pr-bc58cc28d41743a051684a375e8147c902f01749  I'm wondering if it's because of the `const char *const`.  What do you think @ragusaa?

Reply:
The lgtm issue was due to having the ')' in the wrong place.  I am really surprised this did not produce a warning, especially with -Wall -Wextra.

As far as the const char * const, that sets the pointer itself, as well as the data, const.